src/aiokatcp/sensor.py

```diff
@@ -714,3 +748,127 @@ def copy(self) -> Dict[str, Sensor]:
+    ) -> None:
+        super().__init__(
+            sensor_type=sensor_type,
+            name=name,
+            description=description,
+            units=units,
+            auto_strategy=auto_strategy,
+            auto_strategy_parameters=auto_strategy_parameters,
+        )
+        self.target = target
+        for sensor in self.target.values():
+            if self.filter_aggregate(sensor):
+                sensor.attach(self._update_aggregate)
+
+        self.target.add_add_callback(self._sensor_added)
+        self.target.add_remove_callback(self._sensor_removed)
+        self._update_aggregate(None, None, None)  # type: ignore
+
+    def __del__(self):
+        self.target.remove_add_callback(self._sensor_added)
+        self.target.remove_remove_callback(self._sensor_removed)
+        for sensor in self.target.values():
+            if sensor is not self:
+                sensor.detach(self._update_aggregate)
+
+    @abstractmethod
+    def update_aggregate(
+        self,
+        updated_sensor: Optional[Sensor[_U]],
+        reading: Optional[Reading[_U]],
+        old_reading: Optional[Reading[_U]],
+    ) -> Reading[_T]:
+        """Update the aggregated sensor.
+
+        The user is required to override this function, which must return the
+        updated :class:`Reading` (i.e. value, status and timestamp) which will
+        be reflected in the `Reading` of the aggregated sensor.
+
+        Parameters
+        ----------
+        updated_sensor
+            The sensor in the target :class:`SensorSet` which has changed in
+            some way.
+        reading
+            The current reading of the `updated_sensor`. This is `None` if the
+            sensor is being removed from the set.
+        old_reading
+            The previous reading of the `updated_sensor`. This is `None` if the
+            sensor is being added to the set.
+
+        Returns
+        -------
+        Reading
+            The reading (value, status, timestamp) that should be shown by the
+            `AggregatedSensor` as a result of the change.
+        """
+        pass
+
+    def filter_aggregate(self, sensor: Sensor) -> bool:
+        """Check that a sensor isn't this aggregate sensor.
+
+        Users can override this function to exclude certain categories of
+        sensors, such as other aggregates, to prevent circular references.
+
+        Returns
+        -------
+        bool
+            True if `sensor` should be included in calculation of the
+            aggregate, False if not.
+        """
+        return sensor is not self
+
+    def _update_aggregate(
+        self,
+        updated_sensor: Optional[Sensor[_U]],
+        reading: Optional[Reading[_U]],
+        old_reading: Optional[Reading[_U]],
+    ) -> None:
+        updated_reading = self.update_aggregate(updated_sensor, reading, old_reading)
+        self.set_value(updated_reading.value, updated_reading.status, updated_reading.timestamp)
```

Comment:
I think it got lost in one of the other resolved threads, but my suggestion is to make `update_aggregate` return `Optional[Reading[_T]]`, then
```suggestion
        if updated_reading is not None:
            self.set_value(updated_reading.value, updated_reading.status, updated_reading.timestamp)
```
That allows `update_aggregate` to decide that a change is irrelevant and not update the reading (for some sampling strategies, setting the value to the same value still triggers an update to subscribers), so there is a difference.